### PR TITLE
FINERACT-823 Gradle modernizer plugin

### PR DESCRIPTION
**Pull request by percyashu**
## Description
Describe the changes made and why they were made. Ignore if these details are present on the associated Jira ticket https://issues.apache.org/jira/browse/FINERACT-823

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [ ] API documentation at https://github.com/apache/fineract/blob/develop/api-docs/apiLive.htm has been updated with details of any API changes.

- [ ] Integration tests have been created/updated for verifying the changes made.

- [ ] All Integrations tests are passing with the new commits.

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Comment by awasum** [comment]
Thanks for this PR @percyashu ...Next time..before sending a PR, Make sure to run the integration tests locally first using ./gradlew clean integrationTest.. When all passes then can you send a PR... I see this PR has a failure somewhere..Will you like to investigate and find out why? Then update the PR.

**Comment by awasum** [comment]
@percyashu Please..make sure you complete 1 issue before moving to the next one. THere is no point opening more than 1 PR at a time and try to work on all at same time. In general, thats usually a bad idea especially for new programmers. Finish one, then pick another and you find that you make more progress that way. Leave the other PRs you have opened like that and take one of them to fix before moving to the next.

**Comment by percyashu** [comment]
@awasum I did run ./gradlew clean integrationTest and FixedDepositTest does not fail.

**Comment by percyashu** [comment]
@awasum or @vorburger could you review this.

**Comment by awasum** [comment]
@percyashu , Please squash these 3 commits into 1.

**Comment by percyashu** [comment]
@awasum OK I will do that

**Comment by percyashu** [comment]
@awasum I have squash them.

File: fineract-provider/build.gradle [hunk 1]
```diff
@@ -298,6 +300,22 @@ if (project.hasProperty('security') && project.getProperty('security') == 'oauth
     }
 }
 
+modernizer {
+    includeTestClasses = false
+    failOnViolations = false
+
+    violationLogLevel="warn"
```
Comment (vorburger) [inline fineract-provider/build.gradle]: Thanks for this contribution, I love it! Seeing that this only adds additional warn logs during the build, I fear everyone will just ignore this, and it won't really add any value... (This is why Checkstyle and SpotBugs are "enforced" now; if we don't do that, it's pretty much pointless.) I think what would be much more useful is if this was set to fail on violation, at violation log level error. I suspect that probably will cause some failure, but we should fix those...

**Review by vorburger — CHANGES_REQUESTED** [review]

File: fineract-provider/build.gradle (hunk 1 above)
Comment (awasum) [reply fineract-provider/build.gradle]: Ping @percyashu , will u like o address this so it can be merged by @vorburger ?

Comment (percyashu) [reply fineract-provider/build.gradle]: I will make the changes and fix the violation.

**Comment by percyashu** [comment]
@vorburger I resolved all violation except does pertaining to replacing Joda.time with Java.time which will be handle in a PR to issue https://issues.apache.org/jira/projects/FINERACT/issues/FINERACT-826.

**Review by xurror — APPROVED** [review]
LGTM

**Review by vorburger — APPROVED** [review]

**Pull request merged**